projects/plugins/boost/app/modules/cache/Page_Cache.php

```diff
@@ -36,6 +40,38 @@ public function __construct() {
 
 	public function setup() {
 		Garbage_Collection::setup();
+
+		add_action( 'jetpack_boost_module_status_updated', array( $this, 'handle_module_status_updated' ), 10, 2 );
+		add_action( 'jetpack_boost_critical_css_invalidated', array( $this, 'invalidate_cache' ) );
+		add_action( 'jetpack_boost_critical_css_generated', array( $this, 'invalidate_cache' ) );
+	}
+
+	/**
+	 * Handles the module status updated event.
+	 *
+	 * @param string $module_slug The slug of the module that was updated.
+	 */
+	public function handle_module_status_updated( $module_slug, $status ) {
+		// Get a list of modules that can change the HTML output.
+		$output_changing_modules = Modules_Index::get_modules_implementing( Changes_Output::class );
+
+		// Special case: don't clear when enabling Critical or Cloud CSS, as they will
+		// be handled after generation.
+		if ( $status === true ) {
+			unset( $output_changing_modules['critical_css'] );
+			unset( $output_changing_modules['cloud_css'] );
+		}
```

Comment:
This is fine, but I do wonder if the interface we want is actually `Affects_Speed_Score` so that this wouldn't be necessary 🤔

Reply:
It does change things when we turn it of instantly. Turning on doesn't impact it immediately. If we want to differentiate it with interfaces, we may end up with: Modifies_HTML_On_Enable and Modifies_HTML_On_Disable. Which now start to look like too specific separation.

Reply:
`Affects_Speed_Score` says nothing about doing it immediately 🤷